Index: openslide-3.3.0/configure.ac
===================================================================
--- openslide-3.3.0.orig/configure.ac
+++ openslide-3.3.0/configure.ac
@@ -47,35 +47,37 @@ AC_SEARCH_LIBS([floor], [m],, AC_MSG_FAI
 AC_SEARCH_LIBS([jpeg_CreateDecompress], [jpeg],,
 					 AC_MSG_FAILURE([cannot find libjpeg]))
 
-AC_MSG_CHECKING([for OpenJPEG])
-dnl AC_CHECK_LIB won't work with the Win32 version of openjpeg
-dnl because of the stdcall calling convention which requires
-dnl configure to read openjpeg.h.
+#AC_MSG_CHECKING([for OpenJPEG])
+#dnl AC_CHECK_LIB won't work with the Win32 version of openjpeg
+#dnl because of the stdcall calling convention which requires
+#dnl configure to read openjpeg.h.
+#
+#old_LIBS="$LIBS"
+#LIBS="-lopenjpeg $LIBS"
+#AC_LINK_IFELSE(
+#  [AC_LANG_SOURCE(
+#[[
+##include <openjpeg.h>
+#int
+#main ()
+#{
+#  const char *ver = opj_version();
+#  return 0;
+#}
+#]])],
+#  openjpeg_ok=yes,
+#  openjpeg_ok=no)
+#LIBS="$old_LIBS"
+#
+#if test "$openjpeg_ok" = yes; then
+#  OPENJPEG_LIBS="-lopenjpeg"
+#  LIBS="$OPENJPEG_LIBS $LIBS"
+#  AC_MSG_RESULT($openjpeg_ok)
+#else
+#  AC_MSG_FAILURE([cannot find OpenJPEG])
+#fi
 
-old_LIBS="$LIBS"
-LIBS="-lopenjpeg $LIBS"
-AC_LINK_IFELSE(
-  [AC_LANG_SOURCE(
-[[
-#include <openjpeg.h>
-int
-main ()
-{
-  const char *ver = opj_version();
-  return 0;
-}
-]])],
-  openjpeg_ok=yes,
-  openjpeg_ok=no)
-LIBS="$old_LIBS"
-
-if test "$openjpeg_ok" = yes; then
-  OPENJPEG_LIBS="-lopenjpeg"
-  LIBS="$OPENJPEG_LIBS $LIBS"
-  AC_MSG_RESULT($openjpeg_ok)
-else
-  AC_MSG_FAILURE([cannot find OpenJPEG])
-fi
+PKG_CHECK_MODULES(OPENJPEG, [libopenjpeg1])
 
 AC_SEARCH_LIBS([TIFFOpen], [tiff],, AC_MSG_FAILURE([cannot find libtiff]))
 
Index: openslide-3.3.0/Makefile.am
===================================================================
--- openslide-3.3.0.orig/Makefile.am
+++ openslide-3.3.0/Makefile.am
@@ -12,7 +12,7 @@ ACLOCAL_AMFLAGS=-I m4
 
 lib_LTLIBRARIES = src/libopenslide.la
 
-src_libopenslide_la_LIBADD = $(GLIB2_LIBS) $(CAIRO_LIBS) $(LIBXML2_LIBS)
+src_libopenslide_la_LIBADD = $(GLIB2_LIBS) $(CAIRO_LIBS) $(LIBXML2_LIBS) $(OPENJPEG_LIBS)
 
 src_libopenslide_la_SOURCES = src/openslide.c \
 	src/openslide-cache.c src/openslide-tilehelper.c \
@@ -34,7 +34,7 @@ endif
 .rc.lo:
 	$(AM_V_GEN)$(LIBTOOL) $(AM_V_lt) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --tag=RC --mode=compile $(RC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) -i "$<" -o "$@"
 
-src_libopenslide_la_CPPFLAGS = -pedantic -D_OPENSLIDE_BUILDING_DLL $(GLIB2_CFLAGS) $(CAIRO_CFLAGS) $(LIBXML2_CFLAGS) -DG_LOG_DOMAIN=\"Openslide\" -I$(top_srcdir)/src
+src_libopenslide_la_CPPFLAGS = -pedantic -D_OPENSLIDE_BUILDING_DLL $(GLIB2_CFLAGS) $(CAIRO_CFLAGS) $(LIBXML2_CFLAGS) $(OPENJPEG_CFLAGS) -DG_LOG_DOMAIN=\"Openslide\" -I$(top_srcdir)/src
 
 src_libopenslide_la_LDFLAGS = -version-info 3:0:3 -no-undefined
 
